### ISSUE TITLE
Multipart upload does not work 

### ISSUE THREAD

**Issue by anhkieet** [aws/aws-sdk-ios#946]
### To help us solve your problem better, please answer the following list of questions.

* What service are you using?
AWSS3
* In what version of SDK are you facing the problem?
2.6.19
* Is the issue limited to Simulators / Actual Devices?
No
* Can your problem be resolved if you bump to a higher version of SDK?
No, this is the highest version
* Is this problem related to specific iOS version?
 Unknown, I'm on 10.2.1
* How are you consuming the SDK? CocoaPods / Carthage / Prebuilt frameworks?
CocoaPods, Xcode 9.3.1, Swift 4
* Can you give us steps to reproduce with a minimal, complete, and verifiable example? Please include any specific network conditions that might be required to reproduce the problem.

Here is my code with multipart upload
```
func multipartUpload(fileUrl: URL, bucketName: String, key: String) {
//        print("Url: \(fileUrl.absoluteString)")
        let expression = AWSS3TransferUtilityMultiPartUploadExpression()
        expression.progressBlock = {(task, progress) in
            DispatchQueue.main.async(execute: {
                // Do something e.g. Update a progress bar.
                print("progress: \(progress.fractionCompleted)")
            })
        }
        expression.setValue("public-read", forRequestHeader: "x-amz-acl")
        
        var completionHandler: AWSS3TransferUtilityMultiPartUploadCompletionHandlerBlock?
        completionHandler = { (task, error) in
            DispatchQueue.main.async(execute: {
                // Do something e.g. Alert a user for transfer completion.
                // On failed uploads, `error` contains the error object.
                if let err = error {
                    print("Upload failed with error: (\(err.localizedDescription))")
                } else {
                    print("Upload success")
                }
            })
        }
        
        let transferUtility = AWSS3TransferUtility.default()
        
        transferUtility.uploadUsingMultiPart(fileURL: fileUrl, bucket: bucketName, key: key, contentType: "image/jpg", expression: expression, completionHandler: completionHandler).continueWith { (task) -> Any? in
            
            if let error = task.error {
                print("Upload failed with error: (\(error.localizedDescription))")
            }
            
            if let result = task.result {
                let url = AWSS3.default().configuration.endpoint.url
                let publicURL = url?.appendingPathComponent(result.bucket).appendingPathComponent(result.key)
                if let absoluteString = publicURL?.absoluteString {
                    print("Uploaded to:\(absoluteString)")
                }
            }

            return nil
        }
    }
```

But it does not work. This is a log:

```
Url: file:///var/mobile/Containers/Data/Application/663D520C-0711-439F-A9E5-2F881CB29AF3/Documents/test3.jpg
Uploaded to:https://s3-ap-southeast-1.amazonaws.com/davidbui/test3.jpg
```

Completion handler and progress blocks don't work. I have tried it many times but it still does not work

Can you show me what I did wrong? Thanks!

**Comment by cbommas** [comment]
hello @anhkieet 

Sorry to hear that you are having issues. I have a question, just so that I can debug this further: have you been able to get non multipart uploads to work? 

Also, there is a sample app available at https://github.com/awslabs/aws-sdk-ios-samples/tree/master/S3TransferUtility-Sample. This currently only has a non-multipart example, but once we get this working, I can walk you through the steps needed to convert it to multipart.

**Comment by anhkieet** [comment]
Hi @cbommas 

Yes, non-multipart upload functions work well. Only multipart upload does not work

**Comment by cbommas** [comment]
hi @anhkieet 

I dug into this and found a bug in the instantiation logic for the TransferUtility. This will be fixed in the next release of the SDK.

To unblock you immediately, make the following changes on the TransferUtility instantiation logic

        let serviceConfiguration = AWSServiceManager.default().defaultServiceConfiguration
        let transferUtilityConfiguration = AWSS3TransferUtilityConfiguration()
         
        AWSS3TransferUtility.register(
            with: serviceConfiguration,
            transferUtilityConfiguration: transferUtilityConfiguration,
            forKey: "myTransferUtility"
        )

      let transferUtility = AWSS3TransferUtility.s3TransferUtility(forKey: "myTransferUtility")

**Comment by cbommas** [comment]
@anhkieet 

Thank you for confirming. Please feel free to reopen this thread if you run into further issues.

**Comment by cbommas** [comment]
@anhkieet 
The underlying bug that was causing the issue has also been fixed and included in the latest rev, 2.6.20.
